packages/rx_bloc_cli/mason_templates/bricks/feature_onboarding/__brick__/lib/feature_onboarding/ui_components/phone_number_picker.dart

```diff
@@ -0,0 +1,191 @@
+
+class PhoneNumberPicker extends StatelessWidget {
+  const PhoneNumberPicker({
+    this.countryCodeTextStyle,
+    this.countryCodeBgColor = const Color(0xFFE1E1E1),
+    this.phoneNumberHint = 'XX XXX XXXX',
+    this.countryCodeHint = 'XXX',
+    super.key,
+  });
+
+  /// The text style of the country code text
+  final TextStyle? countryCodeTextStyle;
+
+  /// Background color of the country code segment
+  final Color countryCodeBgColor;
+
+  /// The hint text of the country code input
+  final String countryCodeHint;
+
+  /// The hint text of the phone number input
+  final String phoneNumberHint;
+
+  static const double _borderRadius = 16.0;
+
+  @override
+  Widget build(BuildContext context) => Padding(
+        padding: const EdgeInsets.symmetric(horizontal: 8),
+        child: Column(
+          children: [
+            Row(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: [
+                RxBlocBuilder<OnboardingPhoneBlocType, CountryCodeModel?>(
+                  state: (bloc) => bloc.states.countryCode,
+                  builder: (context, countryCodeSnapshot, bloc) =>
+                      _buildPhoneNumberPrefix(
+                    context,
+                    countryCodeSnapshot.data,
+                  ),
+                ),
+                RxBlocBuilder<OnboardingPhoneBlocType, String>(
+                  state: (bloc) => bloc.states.phoneNumber,
+                  builder: (context, numberSnapshot, bloc) => Expanded(
+                      child: _buildPhoneNumberInput(
+                    context,
+                    numberSnapshot.data ?? '',
+                    phoneNumberHint,
+                  )),
+                ),
+              ],
+            ),
+            RxBlocBuilder<OnboardingPhoneBlocType, ErrorModel?>(
+              state: (bloc) => bloc.states.validationErrors,
+              builder: (context, validationErrorSnapshot, bloc) =>
+                  validationErrorSnapshot.data != null
+                      ? Padding(
+                          padding: const EdgeInsets.only(top: 16),
+                          child: Align(
+                            alignment: Alignment.centerLeft,
+                            child: Text(
+                              _translateError(
+                                  context, validationErrorSnapshot.data!),
+                              style: context.designSystem.typography.h2Reg16
+                                  .copyWith(
+                                color: context.designSystem.colors.errorColor,
+                              ),
+                            ),
+                          ),
+                        )
+                      : const SizedBox.shrink(),
+            ),
+          ],
+        ),
+      );
+
+  /// region Builders
+
+  Widget _buildPhoneNumberPrefix(
+    BuildContext context,
+    CountryCodeModel? selectedCountryCode,
+  ) =>
+      TextButton(
```

Comment:
Please use the common ui components from the base folder instead of implementing custom widgets. This will ensure cosistency

Reply:
Done.